src/v/pandaproxy/parsing/httpd.h

```diff
@@ -125,4 +128,27 @@ inline json::serialization_format content_type_header(
     return fmt;
 }
 
+inline std::pair<ss::sstring, ss::sstring>
+basic_auth_header(const seastar::httpd::request& req) {
+    auto auth_header = req.get_header("authorization");
+    if (auth_header.substr(0, 5) != "Basic") {
+        throw ss::httpd::bad_request_exception(
+          "Unsupported Authorization method");
+    }
+
+    // Minimal length: Basic, a space, 1 or more bytes
+    if (auth_header.size() < 7) {
+        throw ss::httpd::bad_request_exception(
+          "Malformed Authorization header");
+    }
+
+    // str needs to be base64 in final PR
+    auto raw_str = auth_header.substr(6);
+    auto colon = raw_str.find(":");
+    ss::sstring username{raw_str.substr(0, colon)};
+    ss::sstring userpass{raw_str.substr(colon + 1)};
```

Comment:
I think the checks here should be a little stronger, I guess username and password should be not empty.